File: iOS/Code/Xamarin/bugTrap/bugTrapKit/Trackers/Pivotal/PivotalDetail.cs
namespace bugTrapKit.Pivotal
{
	public enum PivotalDetail
	{
		Images,
		Project,
		Name,
		Description,
		StoryType,
		Deadline,
		Owners,
		Labels,
		Followers
	}

	public static class PivotalDetailExtensions
	{
		public static string SettingsKey (this PivotalDetail detail)
		{
			switch (detail) {
			case PivotalDetail.Images:
				return "PivotalDetailImages";
			case PivotalDetail.Project:
				return "PivotalDetailProject";
			case PivotalDetail.Name:
				return "PivotalDetailName";
			case PivotalDetail.Description:
				return "PivotalDetailDescription";
			case PivotalDetail.StoryType:
				return "PivotalDetailStoryType";
			case PivotalDetail.Deadline:
				return "PivotalDetailDeadline";
			case PivotalDetail.Owners:
				return "PivotalDetailOwners";
			case PivotalDetail.Labels:
				return "PivotalDetailLabels";
			case PivotalDetail.Followers:
				return "PivotalDetailFollowers";
			default:
				return null;
			}
		}

		public static bool IsPersonDetail (this PivotalDetail detail)
		{
			switch (detail) {
			case PivotalDetail.Owners:
			case PivotalDetail.Followers:
				return true;
			default:
				return false;
			}
		}
	}
}

File: iOS/Code/Swift/bugTrap/bugTrapKit/Trackers/Pivotal/PivotalDetail.swift
//
//  PivotalDetail.swift
//  bugTrap
//
//  Created by Colby L Williams on 10/19/14.
//  Copyright (c) 2014 bugTrap. All rights reserved.
//

import Foundation

enum PivotalDetail: Int, TrackerDetailType {
	case Images = 0, Project, Name, Description, StoryType, Deadline, Owners, Labels, Followers

	func isPersonDetail () -> Bool {
		switch self {
		case .Images, .Project, .Name, .Description, .StoryType, .Deadline, .Labels: return false
//		case .Owners: return true
		case .Owners, .Followers: return true
		}
	}

		var settingsKey: String {
		switch self {
			case .Images: return "PivotalDetailImages"
			case .Project: return "PivotalDetailProject"
			case .Name: return "PivotalDetailName"
			case .Description: return "PivotalDetailDescription"
			case .StoryType: return "PivotalDetailStoryType"
			case .Deadline: return "PivotalDetailDeadline"
			case .Owners: return "PivotalDetailOwners"
			case .Labels: return "PivotalDetailLabels"
			case .Followers: return "PivotalDetailFollowers"
		}
	}
}